client/src/containers/MultiFactorApp.js

```diff
@@ -60,12 +88,21 @@ class MultiFactorApp extends Component {
     }
 
     const { registeredMethods } = schema;
+    const showRegister = loginCompleted || !registeredMethods.length;
 
-    if (!registeredMethods.length) {
-      return <Register {...schema} />;
-    }
-
-    return <Login {...schema} onCompleteLogin={this.handleCompleteLogin} />;
+    return (
+      <div>
+        {title && <h1>{title}</h1>}
+        {showRegister && <Register {...schema} onSetTitle={this.handleSetTitle} />}
+        {showRegister || (
```

Comment:
I think this might be tidier if it were written like this:

```jsx
return (
  <div>
    { ... }
    { this.renderRegister() }
    { this.renderLogin() }
  </div>
);
```

Then you can put the logic for whether to render anything at the top of those methods.

Thoughts?

Reply:
> Thoughts?
I wish we had `v-if` and `v-else` 😆 . I agree it's kind of ugly. I'm often torn about what's best with putting render conditionals in `renderThing` - mostly because in some cases you want "else" logic. I'll do it though - I agree it's cleaner. 🙂